app/services/okta_auth_service.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class OktaAuthService < AuthServiceBase
+  LEEWAY = 0
+  base_uri ENV.fetch("OKTA_SERVER_URL")
+
+  class << self
+    private
+
+    def decode_token(access_token)
+      JWT.decode(access_token, nil, false).first
+    end
+
+    def primary_key
+      :sso_guid
+    end
+
+    def expected_fields
+      %w[exp cid iss]
+    end
+
+    def validate_token!(_access_token, decoded_token)
+      if decoded_token["exp"] < LEEWAY.seconds.ago.to_i
+        raise self::FailedAuthentication, "Expired access_token."
+      end
+      unless decoded_token["cid"].in?(ENV["OKTA_SERVER_AUDIENCE"].split(","))
+        raise self::FailedAuthentication, "Invalid access_token cid."
+      end
+      unless decoded_token["iss"] == ENV["OKTA_SERVER_PATH"]
+        raise self::FailedAuthentication, "Invalid issuer. Expected #{ENV["OKTA_SERVER_PATH"]}, received #{decoded_token["iss"]}"
+      end
+    end
+
+    def remote_user_id(decoded_token)
+      # noop since remote_user_id for okta is ssoguid given in the userinfo call
+    end
+
+    # manually implement setup_user to pull ssoguid from usre_atts instead of using remote_user_id
```

Comment:
I think what you mean is

```suggestion
    # reimplement setup_user to search on sso_guid (from user_atts) instead of using remote_user_id
```